### PR TITLE
"Feat:Create Market API ( Initial code Review Changes)"

### PR DESCRIPTION
**Pull request by pgsaihari**
Implemented the changes suggested as per the initial code review

### CODE REVIEW

File: API/Controllers/MarketController.cs [hunk 1]
```diff
@@ -18,27 +17,28 @@ public MarketController(IMediator mediator)
...
-            {
-                return Ok(new { message = result });
-            }
+            var marketId = await _mediator.Send(command);
+            return CreatedAtAction(nameof(GetById), new { id = marketId }, marketId);
+        }
 
-            return BadRequest(new { error = result });
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetById(int id)
```
Comment (fakrudheenvp) [inline API/Controllers/MarketController.cs]: Make it as GetMarketById

File: API/Controllers/MarketController.cs [hunk 2]
```diff
@@ -18,27 +17,28 @@ public MarketController(IMediator mediator)
...
+        {
+            var market = await _mediator.Send(new GetMarketByIdQuery { Id = id });
+            return market != null ? Ok(market) : NotFound();
         }
 
-     /*   [HttpGet("{id}")]
-        public async Task<IActionResult> GetMarketById(int id)
+        [HttpGet]
+        public async Task<IActionResult> GetAll()
```
Comment (fakrudheenvp) [inline API/Controllers/MarketController.cs]: Make it as GetAllMarkets

File: Application/Requests/MarketRequests/CreateMarketCommadHandler.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        private readonly AppDbContext _context;
+
+        public CreateMarketCommandHandler(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<int> Handle(CreateMarketCommand request, CancellationToken cancellationToken)
+        {
+            var market = new Market
```
Comment (fakrudheenvp) [inline Application/Requests/MarketRequests/CreateMarketCommadHandler.cs:57]: pls write LLD

File: Application/Requests/MarketRequests/GetAllMarketsQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        private readonly AppDbContext _context;
+
+        public GetAllMarketsQueryHandler(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<Market>> Handle(GetAllMarketsQuery request, CancellationToken cancellationToken)
+        {
+            return await _context.Markets.ToListAsync(cancellationToken);
```
Comment (fakrudheenvp) [inline Application/Requests/MarketRequests/GetAllMarketsQueryHandler.cs:47]: LLD

File: Application/Requests/MarketRequests/CreateMarketCommad.cs [hunk 5]
```diff
@@ -5,13 +5,12 @@
 using System.Text;
 using System.Threading.Tasks;
 
-namespace Application.Markets.Commands
+namespace Application.Requests.MarketRequests
 {
-    public class CreateMarketCommand : IRequest<string>
+    public class CreateMarketCommand : IRequest<int>
     {
         public string MarketName { get; set; }
```
Comment (fakrudheenvp) [inline Application/Requests/MarketRequests/CreateMarketCommad.cs]: make it as just Name, code etc

**Review by fakrudheenvp — CHANGES_REQUESTED** [review]

File: Application/Requests/MarketRequests/CreateMarketCommad.cs (hunk 5 above)
Comment (pgsaihari) [reply Application/Requests/MarketRequests/CreateMarketCommad.cs]: `using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Domain.Enums;
using Domain.Enums.Domain.Enums;

namespace Domain.Model
{
    public class Market : AuditableEntity
    {
        public int Id { get; set; }  // Primary Key

        [Required]
        [MaxLength(150)]
        public string Name { get; set; }  // Unique

        [Required]
        [MaxLength(2)]
        public string Code { get; set; }  // Unique, 2 letters

        [Required]
        [MaxLength(50)]
        public string LongMarketCode { get; set; }  // Optional field for long code

        [Required]
        public Region Region { get; set; }  // Enum for Region

        [Required]
        public SubRegion SubRegion { get; set; }  // Enum for SubRegion

        // Navigation property for related MarketSubGroups
        public List<MarketSubGroup> MarketSubGroups { get; set; } = new List<MarketSubGroup>();
    }
}
`

File: Application/Requests/MarketRequests/GetAllMarketsQueryHandler.cs (hunk 4 above)
Comment (pgsaihari) [reply Application/Requests/MarketRequests/GetAllMarketsQueryHandler.cs:47]: LLD Added

**Review by fakrudheenvp — APPROVED** [review]

**Pull request merged**